### PR TITLE
feat: add plugins-benchmark-pr.yml

### LINKED ISSUE #89: Benchmark workflow
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

We should have a nice benchmark workflow for use in different repos.

While writing the benchmark workflow of fsj, i saw some issues, which should be tackled:

- [ ] the pr comment action is using node 12 and results in warnings. I am pretty sure we can use octokit/request-action instead
- [x] we should use the new way to handle outputs, see https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/ . I tried to use it, but there a formatting issues which dont work with the new way of outputting information.
- [ ] we should think of a way to handle a matrix strategy for the node version. Input variables can not be array. So we should check if you are just pass a comma separated list, where we slab brackets around and the use the json parse method of the github runner. This we could use for the other workflows here in this repo too. But for fsj, i dont need benchmarks from node14 to node 20. I just need one, and thats the lts.
- [ ] we need to investigate a nice way to create the pull request body.
- [ ] we should use the default_branch information from the event instead of hardcoding 'master' or 'main'. Maybe use an input variable, where default is the default_branch? 

### Motivation

_No response_

### Example

_No response_

### PR DESCRIPTION
**Pull request by Uzlopak**
Resolves #89 

You can see an example output here: https://github.com/Uzlopak/fast-json-stringify/pull/4#issuecomment-1666122071

Can be integrated like this:

```yml
name: Benchmark PR

on:
  pull_request_target:
    types: 
      - labeled

jobs:
  benchmark:
    if: ${{ github.event.label.name == 'benchmark' }}
    uses: fastify/workflows/.github/workflows/plugins-benchmark-pr.yml@main
    with:
      npm-script: bench

  remove-label:
    if: "always()"
    needs: 
      - benchmark
    runs-on: ubuntu-latest
    steps:
      - name: Remove benchmark label
        uses: octokit/request-action@v2.x
        id: remove-label
        with:
          route: DELETE /repos/{repo}/issues/{issue_number}/labels/{name}
          repo: ${{ github.event.pull_request.head.repo.full_name }}
          issue_number: ${{ github.event.pull_request.number }}
          name: benchmark
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

```

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by RafaelGSS — COMMENTED** [review]
Is it working for forks? IIRC I faced a behavior where `actions/checkout` wasn't able to checkout a branch due to missing `pull_request_target` on the event type

**Comment by Uzlopak** [comment]
Well renamed the reusable wf to plugins-benchmark-pr. 

Well this benchmark wf is only usable with events of type pull_request.

**Comment by Uzlopak** [comment]
If you overall like the workflow, I can add the documentation, where i specify how to use it.

**Review by Fdawgs — APPROVED** [review]

**Review by metcoder95 — APPROVED** [review]

**Review by RafaelGSS — APPROVED** [review]

**Comment by Uzlopak** [comment]
@Fdawgs can you have a look over the readme.md?

**Comment by Fdawgs** [comment]
> @Fdawgs can you have a look over the readme.md?

Will hop on it around 17:30 BST after work 😊

File: README.md [hunk 1]
```diff
@@ -67,6 +66,51 @@ jobs:
 | `license-check-allowed-additional` | false      | string  |           | Provide a semicolon separated list of SPDX-license identifiers that you want to additionally allow. |
 | `lint`                             | false      | boolean | `false`   | Set to `true` to run the `lint` script in a repository's `package.json`.           |
 
+## Benchmark PR-Workflow
```
Comment (Fdawgs) [inline README.md:69]:
```suggestion
## Benchmark PR workflow
```

File: README.md [hunk 2]
```diff
@@ -67,6 +66,51 @@ jobs:
...
+
+The benchmark workflow expects pull_request or pull_request_target events. Most common use is to run the benchmarks when a `benchmark` label is added to the PR.
```
[the remark this answers is not in the document]
Comment (Uzlopak) [reply README.md:71]:
```suggestion
The benchmark workflow expects `pull_request` or `pull_request_target` events. A common use for this workflow is to run benchmarks when a `benchmark` label is added to the PR.
```

**Comment by Uzlopak** [comment]
@Fdawgs 
adapted

**Review by Fdawgs — APPROVED** [review]

**Pull request merged**